pkg/kubectl/cmd/create_role.go

```diff
@@ -179,10 +198,17 @@ func (c *CreateRoleOptions) Validate() error {
 	}
 
 	for _, r := range c.Resources {
-		_, err := c.Mapper.ResourceFor(r)
-		if err != nil {
+		if len(r.Resource) == 0 {
+			return fmt.Errorf("resource must be specified if apiGroup/subresource specified")
+		}
+		if _, err := c.Mapper.ResourceFor(schema.GroupVersionResource{Resource: r.Resource, Group: r.Group}); err != nil {
 			return err
 		}
+		if len(r.SubResource) > 0 {
+			if _, err := c.Mapper.ResourceFor(schema.GroupVersionResource{Resource: r.SubResource}); err != nil {
+				return err
```

Comment:
ResourceFor can filter out some obvious error (like `scal`). although not 100%. I hope to get your advice.